[IMP] developer: add notes on entity references & fix illegal XML

lvsz opened the pull request:
Using `<` inside XML elements will result in a parsing error, an entity reference should be used instead: `&lt;`.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/5159).

AntoineVDV reviewed (approved):
@robodoo r+

AntoineVDV commented on content/developer/tutorials/getting_started/07_basicviews.rst:
```diff
@@ -218,7 +218,18 @@ services *OR* have a unit price which is *NOT* between 1000 and 2000'::
+    ``&lt;`` for ``<`` and ``&amp;`` for ``&``. Other entity references
+    (``&gt;``, ``&apos;`` & ``&quot;``) are optional.
+
+    Example:
+
+        .. code-block:: xml
+
+            <filter name="negative" domain="[('test_val', 'lt;', 0)]"/>
```
No need to over-indent.
```suggestion
    Example:

    .. code-block:: xml

        <filter name="negative" domain="[('test_val', 'lt;', 0)]"/>
```

lvsz replied:
Good catch, fixed it.

AntoineVDV commented on content/developer/tutorials/getting_started/07_basicviews.rst:
```diff
@@ -218,7 +218,18 @@ services *OR* have a unit price which is *NOT* between 1000 and 2000'::
+    ``&lt;`` for ``<`` and ``&amp;`` for ``&``. Other entity references
+    (``&gt;``, ``&apos;`` & ``&quot;``) are optional.
+
+    Example:
```
This could use the [`example`](https://www.odoo.com/documentation/master/contributing/documentation/rst_cheat_sheet.html#example) directive.

lvsz replied:
Didn't know about that one yet, added it.

AntoineVDV commented on content/developer/tutorials/getting_started/07_basicviews.rst:
```diff
@@ -218,7 +218,18 @@ services *OR* have a unit price which is *NOT* between 1000 and 2000'::
+
+            <filter name="negative" domain="[('test_val', 'lt;', 0)]"/>
+
+.. exercise: Add filter and Group By.
```
The missing colon makes the block being treated as a comment.
```suggestion
.. exercise:: Add filter and Group By.
```

lvsz replied:
Thanks, not sure how that colon went missing, but fixed it.

AntoineVDV reviewed (changes requested):
Take care not to update a PR that has been staged for merging (`r+`) because it will also cancel the merge of all the PRs staged in the same time window!

Also, please quash the two commits together, as their scope is the same.

sdegueldre commented on content/developer/tutorials/getting_started/07_basicviews.rst:
```diff
@@ -218,7 +218,18 @@ services *OR* have a unit price which is *NOT* between 1000 and 2000'::
+
+        .. code-block:: xml
+
+            <filter name="negative" domain="[('test_val', 'lt;', 0)]"/>
```
Should be `&lt;`, not just `lt;`

lvsz replied:
Good catch, fixed.

lvsz commented:
Implemented the suggested changes, and squashed the commits.

Sorry about updating while staged for merge, didn't realize it could cause issues like that. Will take it into account in the future.

AntoineVDV commented:
> Sorry about updating while staged for merge, didn't realize it could cause issues like that. Will take it into account in the future.

No problem; it's the kind of thing that you only become aware of when someone points it out^^

AntoineVDV reviewed (approved):
@robodoo r+
